index.js

```diff
@@ -1639,9 +1638,8 @@
             target.prototype.amorphicGetClassName = function () {return target.__name__};
             target.isObjectTemplate = true;
             target.__injections__ = [];
-            target.__templateProps__ = props;
             target.__objectTemplate__ = objectTemplate;
-            var createProps = objectTemplate.getTemplateProperties(props || {});
+            var createProps = objectTemplate.getTemplateProperties(target || {});
```

Comment:
looks like this should be okay, since it's the same reference that gets passed through. If it's a primitive... then we're in trouble anyway.